### PR TITLE
Go updates and cleanups

### PR DESCRIPTION
**Pull request by noise64**
- add test-examples command to help with local testing of the examples (for now go and ts)
- ts and go updates and cleanups

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -50,5 +55,111 @@ pub fn main() {
...
+        example,
+        ExampleParameters {
+            component_name,
+            package_name,
+            target_path,
+        },
+    )
+    .map_err(|e| format!("instantiate failed: {}", e.to_string()))?;
+
+    match &example.language {
```
Comment (vigoo) [inline src/main.rs]: Would be nice if this woudl test the actual `INSTRUCTIONS` and not hardcoded here

Comment (noise64) [reply src/main.rs]: agree, but that will need some cleanups / unification (or some other formalization on what are full / partial / clean commands), let's do it in a follow up (and will extend for the other languages as well)

**Review by vigoo — APPROVED** [review]

**Review by vigoo — CHANGES_REQUESTED** [review]
Wait :)

File: src/cli.rs [hunk 2]
```diff
@@ -54,6 +54,14 @@ pub enum Command {
         #[arg(short, long, alias = "lang")]
         language: Option<GuestLanguage>,
     },
+
+    // Generates and builds the examples components
+    #[command()]
+    TestExamples {
```
Comment (vigoo) [inline src/cli.rs]: We cannot add this here because this `Command` enum is now directly reused in `golem-cli`.

Comment (vigoo) [reply src/cli.rs]: see https://github.com/golemcloud/golem/blob/main/golem-cli/src/oss/command.rs#L54

Comment (noise64) [reply src/cli.rs]: ahh, thanks for catching it

Comment (noise64) [reply src/cli.rs]: extracted to a separate cli [9b29486](https://github.com/golemcloud/golem-examples/pull/49/commits/9b294865114423cef8d16d35fcc211b7ed70dde6)

**Review by vigoo — APPROVED** [review]

**Pull request merged**